src/test/java/io/datareplication/producer/snapshot/SnapshotProducerIntegrationTest.java

```diff
@@ -1,126 +0,0 @@
-import lombok.NonNull;
-import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.Mock;
-import org.mockito.junit.jupiter.MockitoExtension;
-import org.reactivestreams.FlowAdapters;
-import reactor.core.publisher.Flux;
-
-import java.time.Instant;
-import java.util.List;
-import java.util.concurrent.ExecutionException;
-
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
-
-@ExtendWith(MockitoExtension.class)
-class SnapshotProducerIntegrationTest {
-    Timestamp now = Timestamp.of(Instant.now());
-
-    @Test
-    @DisplayName("Should produce snapshot")
-    void shouldProduceSnapshot(@Mock SnapshotPageRepository snapshotPageRepository,
-                               @Mock SnapshotIndexRepository snapshotIndexRepository,
-                               @Mock SnapshotPageUrlBuilder snapshotPageUrlBuilder,
-                               @Mock PageIdProvider pageIdProvider,
-                               @Mock SnapshotIdProvider snapshotIdProvider)
-        throws ExecutionException, InterruptedException {
-        Flux<Entity<SnapshotEntityHeader>> entityFlow = Flux
-            .just("Hello", "World", "I", "am", "a", "snapshot")
-            .map(this::toSnapshotEntity);
-
-        SnapshotId snapshotId = SnapshotId.of("snapshotIdForTest");
-        PageId pageId1 = PageId.of("1");
-        PageId pageId2 = PageId.of("2");
-        PageId pageId3 = PageId.of("3");
-        List<Url> pages = List.of(
-            url(snapshotId, pageId1),
-            url(snapshotId, pageId2),
-            url(snapshotId, pageId3)
-        );
-
-        when(pageIdProvider.newPageId()).thenReturn(pageId1).thenReturn(pageId2).thenReturn(pageId3);
-        when(snapshotIdProvider.newSnapshotId()).thenReturn(snapshotId);
-        when(snapshotPageUrlBuilder.pageUrl(any(SnapshotId.class), any(PageId.class))).thenAnswer(input ->
-            url(input.getArgument(0), input.getArgument(1)));
-
-        SnapshotIndex expectedSnapshotIndex = new SnapshotIndex(snapshotId, now, pages);
-
-        SnapshotProducer snapshotProducer = SnapshotProducer
-            .builder()
-            .pageIdProvider(pageIdProvider)
-            .snapshotIdProvider(snapshotIdProvider)
-            .maxBytesPerPage(2)
-            // TODO: Additional configuration
-            .build(snapshotIndexRepository, snapshotPageRepository, snapshotPageUrlBuilder);
-
-
-        SnapshotIndex result = snapshotProducer
-            .produce(FlowAdapters.toFlowPublisher(entityFlow))
-            .toCompletableFuture()
-            .get();
-
-        assertThat(result).isEqualTo(expectedSnapshotIndex);
-        verify(snapshotIndexRepository).save(expectedSnapshotIndex);
-        // TODO: Add SnapshotPageHeader
-        verify(snapshotPageRepository).save(
```

Comment:
Notice to myself: Restore IntegrationTest!
Use InMememoryRepos, create stable boundary with provider to assert the generated multipart page . Assert generated index json (all files should be placed into the resources folder)

Reply:
I've created a seperate ticket for restoring the integration test.